db/migrate/20220408080838_bigint_primary_key_on_relations.rb

```diff
@@ -0,0 +1,66 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2022 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+class BigintPrimaryKeyOnRelations < ActiveRecord::Migration[6.1]
+  def up
+    change_relations_id(:bigint, 9223372036854775807)
+  end
+
+  def down
+    change_relations_id(:integer, 2147483647)
+  end
+
+  private
+
+  def change_relations_id(type, maxvalue)
+    # We cannot simply use
+    #
+    # ALTER SEQUENCE relations_id_seq AS bigint;
+    #
+    # since that is not supported by PostgreSQL 9.6.
+    execute <<~SQL.squish
```

Comment:
@oliverguenther in here, some detour had to be taken so that the code is still running on 9.6. Let's talk about when to completely drop support for this.

Reply:
What is the minimum version supporting the alter sequence call? It would be fine for me to force users to use a newer version, but we have known users stuck on versions PSQL 11 and 12.

Reply:
I would leave it as it is in this case since the detour is bearable and already done but it would be interesting to know based on your knowledge of existing instances, whether supporting 9.6 is still beneficial.

`ALTER SEQUENCE` is supported [since PG 10](https://www.postgresql.org/docs/10/sql-altersequence.html)